File: app/src/main/java/com/karthik/myweather/utils/BusinessUtils.java
package com.karthik.myweather.utils;

import com.karthik.myweather.data.entities.CityWeather;
import com.karthik.myweather.data.entities.Weather;
import com.karthik.myweather.network.model.ConsolidatedWeather;

import java.util.stream.Collectors;

public class BusinessUtils {

    public CityWeather convertToDatabaseEntiry(int locationIdToSearch, ConsolidatedWeather consolidatedWeather) {
        CityWeather cityWeather = new CityWeather();
        cityWeather.city = new City(){{
            locationId = consolidatedWeather.getLocationId();
            cityName = consolidatedWeather.getCityName();
        }};
        cityWeather.weatherList = consolidatedWeather
                .getConsolidatedWeather()
                .stream()
                .map(locationWeather -> new Weather(){{
                    locationId = locationIdToSearch;
                    date = locationWeather.getDate();
                    minTemperature = locationWeather.getMinTemperature();
                    maxTemperature = locationWeather.getMaxTemperature();
                    windSpeed = locationWeather.getWindSpeed();
                    weatherCode = locationWeather.getWeatherStateAbbr();
                    weatherDescription = locationWeather.getWeatherStateName();
                }})
                .collect(Collectors.toList());
        return cityWeather;
    }

}

File: app/src/main/java/com/karthik/myweather/utils/BusinessUtils.kt
package com.karthik.myweather.utils

import com.karthik.myweather.data.entities.City
import com.karthik.myweather.data.entities.CityWeather
import com.karthik.myweather.data.entities.Weather
import com.karthik.myweather.network.model.ConsolidatedWeather
import java.util.stream.Collectors
import javax.inject.Inject

class BusinessUtils {
    fun convertToDatabaseEntiry(locationIdToSearch: Int, consolidatedWeather: ConsolidatedWeather): CityWeather {
        val cityWeather = CityWeather()
        val city = City(consolidatedWeather.locationId, consolidatedWeather.cityName)
        cityWeather.city = city
        cityWeather.weatherList = consolidatedWeather
                .consolidatedWeather
                ?.map { (_, weatherStateName, weatherStateAbbr, minTemperature, maxTemperature, windSpeed, date) ->
                    Weather(locationIdToSearch,
                            date!!,
                            minTemperature,
                            maxTemperature,
                            windSpeed,
                            weatherStateAbbr,
                            weatherStateName)
                }
        return cityWeather
    }
}